src/chart/sankey/SankeySeries.js

```diff
@@ -25,26 +25,55 @@
 import SeriesModel from '../../model/Series';
 import createGraphFromNodeEdge from '../helper/createGraphFromNodeEdge';
 import {encodeHTML} from '../../util/format';
+import Model from '../../model/Model';
 
 var SankeySeries = SeriesModel.extend({
 
     type: 'series.sankey',
 
     layoutInfo: null,
 
+    levelModels: null,
+
     /**
      * Init a graph data structure from data in option series
      *
      * @param  {Object} option  the object used to config echarts view
      * @return {module:echarts/data/List} storage initial data
      */
-    getInitialData: function (option) {
+    getInitialData: function (option, ecModel) {
         var links = option.edges || option.links;
         var nodes = option.data || option.nodes;
+        var levels = option.levels;
+        var levelModels = this.levelModels = {};
+
+        for (var i = 0; i < levels.length; i++) {
+            levelModels[levels[i].depth] = new Model(levels[i], this, ecModel);
```

Comment:
Should we check like:
```js
if (__DEV__) {
    if (levels[i].depth == null) {
        throw new Error('levels[i].depth is mandatory')
    }
}
```
Just afraid of confusing with `treemap` and `sunburst` levels where array index is used as the key.